File: geolocationApp/src/main/java/com/example/geolocalisation/Building.java
package com.example.geolocalisation;

import java.util.List;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;
public class Building {


    @SerializedName("rooms_results")
    @Expose
    private List<Room> rooms = null;

    public Building(List<Room> rooms)
    {
        this.rooms = rooms;
    }

    public List<Room> getRooms()
    {
        return rooms;
    }
}

File: geolocationApp/src/main/java/com/example/geolocalisation/Building.kt
package com.example.geolocalisation

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

class Building(rooms: List<Room>?) {
    @SerialName("salles_results")
    @Serializable
    private var rooms: List<Room>? = null

    init {
        this.rooms = rooms
    }

    fun getRooms(): List<Room>? {
        return rooms
    }
}
